R/copernicus_article.R

```diff
@@ -12,21 +12,23 @@
 #'
 #' An number of required and optional manuscript sections, e.g. \code{acknowledgements}, \code{competinginterests}, or \code{authorcontribution}, must be declared using the respective properties of the R Markdown header - see skeleton file.
 #'
-#' \strong{Version:} Based on copernicus_package.zip in the version 5.3, 18 February 2019, using \code{copernicus.cls} in version 8.82.
+#' \strong{Version:} Based on \code{copernicus_package.zip} in the version 6.2, 15 January 2021, using \code{copernicus.cls} in version 9.25.
 #'
-#' \strong{Copernicus journal abbreviations:} You can use the function \code{copernicus_journal_abbreviations()} to get the journal abbreviation for all journals supported by the copernicus article template.
+#' \strong{Copernicus journal abbreviations:} You can use the function \code{copernicus_journal_abbreviations()} to get the journal abbreviation for all journals supported by the Copernicus article template.
 #'
 #' \strong{Important note:} The online guidelines by Copernicus are the official resource.
 #' Copernicus is not responsible for the community contributions made to support the template in this package.
-#' Copenicus converts all typeset TeX files into XML, the expressions and markups have to be highly standardized.
+#' Copernicus converts all typeset TeX files into XML, the expressions and markups have to be highly standardized.
 #' Therefore, please keep the following in mind:
 #'
 #' \itemize{
 #'   \item Please provide only one figure file for figures with several panels, and please do not use \code{\\subfloat} or similar commands.
 #'   \item Please use only commands in which words, numbers, etc. are within braces (e.g. \code{\\textrm{TEXT}} instead of \code{{\\rm TEXT}}).
 #'   \item For algorithms, please use the syntax given in template.tex or provide your algorithm as a figure.
 #'   \item Please do not define new commands.
-#'   \item The most commonly used packages (\code{\\usepackage{}}) are integrated in the copernicus.cls. Some other packages often used by the community are defined in template.tex. Please do not insert additional ones in your *.tex file.
+#'   \item Supported packages (\code{\\usepackage{}}) are already integrated in the \code{copernicus.cls}.  Please do not insert additional ones in your \code{*.tex} file.
```

Comment:
OK so now I see: this is the main change ! 

Copernicus does not allow more package to be used - only those already define in the cls file, is it ? 

How do you know which package can be used then ? because I supposed you need them to be installed and unfortunately copernicus.cls is not available in a CTAN package that would also install dependency. 

How can we help user know about that from the Rmd skeleton or the doc ? 

by the way, if this is the rule, you could warn more about that in `copernicus_article()` bc it means `extra_dependencies:` is not allowed as it will add `\usepackage` in preamble. 

Could be a way to avoid later surprises ? 

Also maybe we should add a comment in the rticles's template so that further contributor know that there should be no `\usepackage` command added in this template

What do you think ? 

cc @nuest

Reply:
@cderv, Thank you for your detailed feedback. I always realise again that you have a broader picture in mind than me. 

> Copernicus does not allow more package to be used - only those already define in the cls file, is it ?

Yes, according to @nuest this was probably allowed in the past. I cannot tell. 

> How do you know which package can be used then ? because I supposed you need them to be installed and unfortunately copernicus.cls is not available in a CTAN package that would also install dependency.

Oh, Daniel found already a good solution, I think, in the `template.tex` he had added all supported packages:

```%% \usepackage commands included in the copernicus.cls:
%\usepackage[german, english]{babel}
%\usepackage{tabularx}
%\usepackage{cancel}
%\usepackage{multirow}
%\usepackage{supertabular}
%\usepackage{algorithmic}
%\usepackage{algorithm}
%\usepackage{amsthm}
%\usepackage{float}
%\usepackage{subfig}
%\usepackage{rotating}
```

> How can we help user know about that from the Rmd skeleton or the doc ?

Perhaps the cheap solution is adding this information from the `template.tex` to the `*.Rmd` file?

> by the way, if this is the rule, you could warn more about that in copernicus_article() bc it means extra_dependencies: is not allowed as it will add \usepackage in preamble.

Thanks, I added this to the documentation and I added a `stop()` to the `copernicus_article()` if the user tries something that would not be allowed.. I guess we would not a test too, but here I need your help. I have no idea where I should add this unit test.

>Could be a way to avoid later surprises ?

As written above, we could just add the list of supported packages to the `*.Rmd` file. Perhaps that is enough?

> Also maybe we should add a comment in the rticles's template so that further contributor know that there should be no \usepackage command added in this template

A valuable point. I added such a piece of information to the `template.tex`.

Reply:
Yes, we should add a comment in the template main body to make sure users see that.

Good point about `extra_dependencies` - do you suggest to catch that in the function and put out a `warning()` ?

Reply:
Actually, I already added a hard `stop()` in the code with feasible advice to the user. But perhaps the `stop()` is too much?

Reply:
yes maybe stop is too much... 🤔 
Maybe just a warning as this is really advice to follow if you want to submit your paper to copernicus ? 

We could also add explicitly the argument to the function, to documented it in the help page;, catch a different value than NULL, and stop or warn that it will be set to NULL. Usually, if you have extra dependencies, it will be required to compile so a stop it could be. 

I don't know who would use this template but not published with the editor...

Reply:
Ok, it is now a `warning()`. I did not add another argument, because it would add something, which is basically not wanted anyway by the publisher (means, better keeping documentation concise).